test/common_test.py

```diff
@@ -301,34 +351,76 @@ def check_save_set_no_objects(
         )
 
 
-def check_get_set_bad_ref(
+@pytest.mark.parametrize("set_type", SET_TYPES)
```

Comment:
Nice!

Reply:
After finishing up this gauntlet of PR activity, it might be a good idea to add some of these things to the readme - like updating this list when creating a new Set object for the API to manage.

Reply:
Or maybe some separate developer docs.